indigo_social/models.py

```diff
@@ -1,8 +1,61 @@
 # -*- coding: utf-8 -*-
+import os
+from io import BytesIO
+import sys
+
 from django.contrib.auth.models import User
+from django.core.files import File
+from django.core.files.uploadedfile import InMemoryUploadedFile
 from django.db import models
-from django.db.models.signals import post_save
+from django.db.models.signals import post_save, pre_save
 from django.dispatch import receiver
+from django.urls import reverse
+from PIL import Image
+
+from urllib.request import urlretrieve, urlcleanup
+
+from allauth.account.signals import user_signed_up
+from allauth.socialaccount.models import SocialLogin, SocialAccount
+from allauth.socialaccount.signals import pre_social_login, social_account_added, social_account_updated
+
+
+def user_profile_photo_path(instance, filename):
+    file_name, file_extension = os.path.splitext(filename)
+    return 'avatars/{}'.format(instance.id)
```

Comment:
I've been burned by `instance.id` being None in the past. I suggest check for None and raise `ValueError`.